playbooks/roles/compatibility-layer/tasks/main.yml

```
@@ -0,0 +1,34 @@
+# TODO: when startprefix does not exist
+# use container to install prefix?
+#- include_tasks: install_prefix.yml
+
+- name: Check if a Prefix installation is found at the specified location
+  stat:
+    path: "{{ gentoo_prefix_path }}/usr/bin/emerge"
+  register: emerge
+
+- name: Fail if the given Prefix path does not have emerge
+  fail:
+    msg: |
+         The given Gentoo Prefix path does not seem to be valid,
+         because the emerge command cannot be found in {{ gentoo_prefix_path }}/usr/bin.
+         Please check it and fix (the path to) the Gentoo Prefix installation.
+  when: not emerge.stat.exists
+
+- name: Start transaction
+  command: "cvmfs_server transaction {{ cvmfs_repository }}"
+  when: cvmfs_start_transaction
+
+- block:
+  - include_tasks: add_overlay.yml
+
+  - include_tasks: install_packages.yml
+
+  - name: Publish transaction
+    command: "cvmfs_server publish {{ cvmfs_repository }}"
+    when: cvmfs_publish_transaction
+
+  rescue:
+    - name: Abort transaction
+      command: "cvmfs_server abort {{ cvmfs_repository }}"
+      when: cvmfs_abort_transaction_on_failures
```

```suggestion
      when: cvmfs_start_transaction and cvmfs_abort_transaction_on_failures
```
For completion check that a transaction has been started